client/components/auth-form.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+
+export default class AuthForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      region: '',
+      timeAvailable: ''
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.userRef = React.createRef();
+    this.passRef = React.createRef();
+    this.timeRef = React.createRef();
+    this.regRef = React.createRef();
+  }
+
+  handleChange(event) {
+    const { id, value } = event.target;
+    this.setState({ [id]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { action } = this.props;
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch(`/api/auth/${action}`, req)
+      .then(res => res.json())
+      .then(result => {
+      });
+    this.setState({ username: '', password: '', region: '', timeAvailable: '' });
+    this.userRef.current.value = '';
+    this.passRef.current.value = '';
+    this.timeRef.current.value = 'placeholder';
+    this.regRef.current.value = 'placeholder';
```

Comment:
I'm not really sure what these placeholders are for, but you can set the initial values in your state.